fv3gfs/util/utils.py

```diff
@@ -57,16 +57,24 @@ def ensure_contiguous(maybe_array: Union[np.ndarray, Storage]) -> None:
         raise ValueError("ndarray is not contiguous")
 
 
-def assign_array(
-    to_array: Union[np.ndarray, Storage], from_array: Union[np.ndarray, Storage],
+def assign_array_via_cpu(
+    to_array: Union[np.ndarray, Storage], from_array: Union[np.ndarray, Storage]
```

Comment:
This is not really what the function does, if both arguments are gpu arrays then the assignment isn't done via cpu. Could you revert the name change and update the docstring? This is meant to be a general purpose "safe" assignment between possibly gpu/cpu arrays/storages.

If what you're going for is a performance speedup by trying `array[:] = source` first, you could put the try: except block for that assignment in this routine instead.